tests/integration/test_hyper_transformer.py

```diff
@@ -167,7 +167,7 @@ def test_hypertransformer_default_inputs():
     pd.testing.assert_frame_equal(transformed, expected_transformed)
 
     reversed_datetimes = pd.to_datetime([
-        '2010-01-09 20:34:17.142857216',
```

Comment:
This was actually a bug, in `python 3.7`, the format wasn't being learned, this change fixed it: `datetime_array = data[data.notna()].astype(str).to_numpy()`